### PR TITLE
Added roadmap of html

### LINKED ISSUE #2761: [Feature Request]: Adding HTML roadmap
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

i would like to add HTML roadmap

### Use Case

adding html roadmap

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by dhairyagothi**
## Related Issue
Fix #2761 
## Description
@Ajay-Dhangar i have added typescript file as you said but its showing error page please check code and help me with this

image

[Please include a brief description of the changes or features added]

## Type of PR

- [ ] Bug fix
- [X] Feature enhancement
- [ ] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [ ] I have performed a self-review of my code.
- [ ] I have read and followed the Contribution Guidelines.
- [ ] I have tested the changes thoroughly before submitting this pull request.
- [ ] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have followed the code style guidelines of this project.
- [ ] I have checked for any existing open issues that my pull request may address.
- [ ] I have ensured that my changes do not break any existing functionality.
- [ ] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [ ] I have read the resources for guidance listed below.
- [ ] I have followed security best practices in my code changes.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [ ] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [ ] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [ ] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [ ] [Markdown Guide](https://www.markdownguide.org/)
- [ ] [MDX Documentation](https://mdxjs.com/docs/)
- [ ] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [ ] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [ ] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [ ] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

**Comment by Ajay-Dhangar** [comment]
@dhairyagothi Alright, I'll solve it.

File: src/pages/roadmap/html/index.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+export default function HtmlRoadMap() {
```
Comment (Ajay-Dhangar) [inline src/pages/roadmap/html/index.tsx:3]: replace `HtmlRoadMap` to `HTML`

File: src/components/Roadmap/index.tsx [hunk 2]
```diff
@@ -12,7 +13,7 @@ const techCategories: TechCategory[] = [
     id: 1,
     title: "Frontend",
     technologies: [
-      { id: 1, name: "HTML", link: "#" },
+      { id: 1, name: "HTML", link: "HtmlRoadMap" },
```
Comment (Ajay-Dhangar) [inline src/components/Roadmap/index.tsx:16]: replace `HtmlRoadMap` to `/roadmap/html`

**Comment by dhairyagothi** [comment]
Thanku @Ajay-Dhangar but tell me error so i can also learn and take care from next time

File: src/components/Roadmap/index.tsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import React, { useState } from "react";
 import "./Roadmap.css";
+import HtmlRoadMap from "../../pages/roadmap/html";
```
Comment (Ajay-Dhangar) [inline src/components/Roadmap/index.tsx:3]: remove `import HtmlRoadMap from "../../pages/roadmap/html";` line

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
Your roadmap content could use some improvements. Consider checking out [this roadmap](https://roadmap.sh/) for a more structured and comprehensive approach. Additionally, for basic HTML topics, refer to [this HTML roadmap](https://roadmap.sh/r/html-mastery-bsx5q). These resources can help you enhance the quality and completeness of your roadmap content.

**Comment by Ajay-Dhangar** [comment]
> Thanku @Ajay-Dhangar but tell me error so i can also learn and take care from next time

@dhairyagothi, I shared some suggestions with you. Could you please review and implement them to improve the content?

File: src/components/Roadmap/index.tsx (hunk 3 above)
Comment (dhairyagothi) [reply src/components/Roadmap/index.tsx]: resolved working now

File: src/components/Roadmap/index.tsx (hunk 2 above)
Comment (dhairyagothi) [reply src/components/Roadmap/index.tsx]: resolved

File: src/pages/roadmap/html/index.tsx (hunk 1 above)
Comment (dhairyagothi) [reply src/pages/roadmap/html/index.tsx]: resolved

**Comment by dhairyagothi** [comment]
@Ajay-Dhangar  i have tried making roadmap like roadmap.sh but it is taking time and some more stacks so for sure i will update this soon with a PR 
thank you please consider this for now 

image

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**